gcp/workers/alias/upstream_hierarchy_computation.py

```diff
@@ -0,0 +1,95 @@
+#!/usr/bin/env python3
+# Copyright 2025 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""OSV Upstream relation computation."""
+
+import datetime
+from google.cloud import ndb
+
+import osv
+import osv.logs
+import json
+
+
+def compute_upstream_hierarchy(target_bug: osv.UpstreamGroup,
+                               bug_groups) -> dict[str, set[str]]:
```

Comment:
should this be named all_upstream_group and target_upstream_group? bug_group is a bit confusing imo.

Reply:
nit: Can you also add `ndb.Query` type to bug_groups?

The documentation for the type in the docstring is really nice though!